client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -43,282 +38,269 @@ export async function drag(testId: any, start: any, end: any, lasso = false) {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  const histTestIds = testIds.map((tid: any) => `histogram-${tid}`);
-
-  // these load asynchronously, so we need to wait for each histogram individually,
-  // and they may be quite slow in some cases.
-  // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 2.
-  await waitForAllByIds(histTestIds, { timeout: 4 * 60 * 1000 });
-
-  const allHistograms = await getAllByClass(testclass);
-
-  const testIDs = await Promise.all(
-    allHistograms.map((hist) =>
-      page.evaluate((elem) => elem.dataset.testid, hist)
-    )
-  );
-
-  return testIDs.map((id) => id.replace(/^histogram-/, ""));
-}
-
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getAllCategoriesAndCounts(category: any) {
+export async function getAllCategoriesAndCounts(
+  category: string,
+  page: Page
+): Promise<{ [cat: string]: string }[]> {
   // these load asynchronously, so we have to wait for the specific category.
-  await waitByID(`category-${category}`);
-
-  return page.$$eval(
-    `[data-testid="category-${category}"] [data-testclass='categorical-row']`,
-    (rows) =>
-      Object.fromEntries(
-        rows.map((row) => {
-          // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-          const cat = row
-            .querySelector("[data-testclass='categorical-value']")
-            .getAttribute("aria-label");
-
-          const count = (
-            row.querySelector(
-              "[data-testclass='categorical-value-count']"
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-            ) as any
-          ).innerText;
-
-          return [cat, count];
-        })
-      )
+  const categoryRows = await page
+    .getByTestId(`category-${category}`)
+    .getByTestId("categorical-row")
+    .all();
+
+  return Object.fromEntries(
+    await Promise.all(
+      categoryRows.map(async (row) => {
+        const cat = await row
+          .getByTestId("categorical-value")
+          .getAttribute("aria-label");
+
+        const count = await row
+          .getByTestId("categorical-value-count")
+          .innerText();
+
+        return [cat, count];
+      })
+    )
   );
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function getCellSetCount(num: any) {
-  await clickOn(`cellset-button-${num}`);
-  return getOneElementInnerText(`[data-testid='cellset-count-${num}']`);
+export async function getCellSetCount(
+  num: number,
+  page: Page
+): Promise<string> {
+  await page.getByTestId(`cellset-button-${num}`).click({ force: true });
```

Comment:
I think it might be work adding a comment to explain why `{ force: true }` is used here. I know you verbally told Ronen and I why, but Alec and the rest of the team would still love the reminder 😄 Thank you!

here and throughout

Reply:
I'll add a comment in at the top of the file